### ISSUE TITLE
Unable to reference the Arn of a Kinesis stream for AWS::DynamoDB::GlobalTable for a AWS::Serverless::Function

### ISSUE THREAD

**Issue by Danny-UKDM** [aws-cloudformation/cfn-lint#2132]
*cfn-lint version: 0.54.2*

*Description of issue.*

The linter fails when using the `!GetAtt` function to reference the Arn of a stream from a DynamoDB Global Table. This looks to be because `cfn-lint` is looking specifically for `AWS::DynamoDB::Table` and does not accept `AWS::DynamoDB::GlobalTable`.

Kinesis streams from Global DynamoDB Tables have additional functionality to maintain replicas across all regions, but the records on the stream itself can also function as a regular DynamoDB kinesis stream to trigger Lambda. (some information here: https://aws.amazon.com/blogs/database/how-to-use-amazon-dynamodb-global-tables-to-power-multiregion-architectures/#:~:text=How%20to%20create%20a%20global%20table)

See below for a pseudo cloudformation example of a syntax which should be valid: 

```yaml
  MyGlobalTable:
    Type: AWS::DynamoDB::GlobalTable
    Properties:
      AttributeDefinitions:
      - AttributeName: PK
        AttributeType: S
      KeySchema:
      - AttributeName: PK
        KeyType: HASH
      Replicas:
      - Region: eu-west-1
      StreamSpecification:
        StreamViewType: NEW_AND_OLD_IMAGES

  MyFunction:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: someCodeUri
      Handler: someHandler
      Events:
        Stream:
          Type: DynamoDB
          Properties:
            Stream: !GetAtt MyGlobalTable.StreamArn # <-- this throws an error in cfn-lint
            BatchSize: 1
            StartingPosition: TRIM_HORIZON
```

**Output:** 
```
E3008 Property "EventSourceArn" can Fn::GetAtt to a resource of types [AWS::DynamoDB::Table, AWS::Kinesis::Stream, AWS::Kinesis::StreamConsumer, AWS::SQS::Queue] at Resources/MyFunctionStream/Properties/EventSourceArn/Fn::GetAtt
```

**Expected Output:**

No errors should output.

**Comment by PatMyron** [comment]
https://github.com/aws-cloudformation/cfn-lint/blob/75510c7f091deefa609fd54ba4ba88ad608ff417/src/cfnlint/data/ExtendedSpecs/all/03_value_types/aws_lambda.json#L28-L38
